pkg/extensions/upgrade.go

```diff
@@ -47,6 +47,8 @@ func Upgrade(ctx context.Context, kcli client.Client, prev *ecv1beta1.Installati
 	// diff the extensions
 	diffResult := diffExtensions(prev.Spec.Config.Extensions, in.Spec.Config.Extensions)
 
+	// NOTE: this does not respect order and may add extensions prior to upgrading ones with a lower order.
```

Comment:
order is respected in the `diffExtensions` function per operation though. How should it be respected otherwise?

Reply:
im not sure how it should be respected. but what if an extension needed to be upgraded first because it contained a new crd for a cr that was applied from another extension that was added?